### DISCUSSION TITLE
Stochastic sampling for a function of boundary condition

### DISCUSSION THREAD

**Discussion by purepuremilk-zz** [idaholab/moose#18205] [Q&A Modules: General] [answered]
Dear MOOSE team, 

I am trying to use stochastic tool for sampling the boundary condition of my simulations. Instead of sampling a constant value of BC, I am sampling a function parameter "alpha" in function 'bc_func' for my BC. However, when I am trying to transfer the sampled values from a uniform distribution to the function parameter, I get an error saying:

 "The desired parameter 'The desired parameter 'Functions/bc_func/vals' was not located for the 'stochastic' object, it either does not exist or has not been declared as controllable." 

Could someone please help resolve this issue? 

Follows are more details of my input. For the simulation input, I am using the ParsedFunction for the function of boundary condition: 

`[Functions]
  [./bc_func]
    type = ParsedFunction
    value = sin(alpha*t)+1
    vars = 'alpha'
    vals = 0.05
  [../]
[]`

This ParsedFunction is later called in the boundary condition: 

`[BCs]
  [top]
    type = FunctionDirichletBC
    variable = T
    boundary = 'top'
    function = bc_func
  []
[]`

The simulation runs fine. The error appears when I try to run with stochastic too. In the master input, I have a transfer block: 
`[Transfers]
  [parameters]
    type = SamplerParameterTransfer
    multi_app = runner
    sampler = hypercube
    parameters = 'Materials/constant/prop_values BCs/bottom/value Functions/bc_func/vals Kernels/source/value'
    to_control = 'stochastic'
  []
[]`

**Comment by jiangwen84** [comment, marked as answer]
The error message tells that the `vals` has not been declared as controllable. 

You can use `MultiAppCommandLineControl`  and we discuss that a bit in #18046

**Reply by aeslaughter** [reply]
Another option would be to make this parameter controllable, but requires a change to MOOSE. 
https://mooseframework.inl.gov/moose/syntax/Controls/index.html
